File: 2148.元素计数.py
#
# @lc app=leetcode.cn id=2148 lang=python3
#
# [2148] 元素计数
#

# @lc code=start
from typing import List
class Solution:
    def countElements(self, nums: List[int]) -> int:
        nmax, nmin = max(nums), min(nums)
        return sum(nmin< i < nmax for i in nums)
# @lc code=end

print(Solution().countElements([1,1,1,1,2,2,3]))

File: 2148.元素计数.cpp
/*
 * @lc app=leetcode.cn id=2148 lang=cpp
 *
 * [2148] 元素计数
 */

// @lc code=start
#include<vector>
using namespace std;
class Solution {
public:
    int countElements(vector<int>& nums) {
        int nmax = INT_MIN, nmin = INT_MAX, ret = 0;
        for(auto i: nums){
            nmax = max(nmax, i);
            nmin = min(nmin, i);
        }
        for(auto i:nums){
            ret += nmin < i && i < nmax;
        }
        return ret;
    }
};
// @lc code=end

